openimageio: add --with-python3 and update to 1.6.9.

jaguarondi opened the pull request:
Add the `--with-python3` option to build and install the python3
bindings.

Update to 1.6.9 release.

dpo commented on openimageio.rb:
```diff
@@ -13,28 +13,31 @@ class Openimageio < Formula
   end
 
-  option "with-tests",  "Dowload 95MB of test images and verify Oiio (~2 min)"
+  option "without-opencv", "Build without OpenCV support"
```
This is not necessary. An option to disable `opencv` support will be automatically created because of line 39.

dpo commented:
`field3d` needs to be bumped:
```
==> brew test --verbose homebrew/science/openimageio
==> FAILED
Testing homebrew/science/openimageio
==> Using the sandbox
/usr/bin/sandbox-exec -f /tmp/homebrew20160114-44577-1arikul-0.sb /Users/brewadmin/Homebrew/Cellar/ruby187/1.8.7-p374_1/bin/ruby -W0 -I /usr/local/Library/Homebrew -- /usr/local/Library/Homebrew/test.rb /usr/local/Library/Taps/homebrew/homebrew-science/openimageio.rb --verbose
==> /usr/local/Cellar/openimageio/1.6.9/bin/oiiotool --info /usr/local/Library/Homebrew/test/fixtures/test.jpg
dyld: Library not loaded: /usr/local/lib/libhdf5.9.dylib
  Referenced from: /usr/local/opt/field3d/lib/libField3D.dylib
  Reason: image not found
```

dpo commented on openimageio.rb:
```diff
@@ -13,28 +13,31 @@ class Openimageio < Formula
-  depends_on "freetype"
-  depends_on "openssl"
-  depends_on "giflib" => :optional
+  depends_on "homebrew/science/opencv" => :recommended
```
There's no need to fully qualify science formulas in this tap.

jaguarondi commented:
@dpo : I updated according to your comments. The bump of field3d now generates an error:

```
Error: Failed to import: /usr/local/Library/Taps/homebrew/homebrew-science/field3d-1.6.1_2.el_capitan.bottle.rb
undefined method `bottle' for Formulary::FormulaNamespace76282ebb86e370d8d7c3f84761638de1:Module
==> PASSED
Error: No available formula with the name "undefined method `bottle' for Formulary::FormulaNamespace76282ebb86e370d8d7c3f84761638de1:Module"
```
How should I handle this? I've seen that you bumped mlpack in a separate pull request, is that the way to go?

dpo commented:
Sorry, I wasn't very clear. `field3d` should be bumped in a separate PR, that we'll pull before rebuilding this one.

jaguarondi commented:
OK, thank you. Done in #3225.
Once it will be pulled, is there a way to request a new build other than making a new commit?

dpo commented:
Please rebase your branch against `master`. There's no need to bump `field3d` here.

jaguarondi commented:
Done, thank you for your help.

dpo commented on openimageio.rb:
```diff
@@ -110,16 +121,19 @@ def install
+    system "make", "test" if build.with? "test"
     cd "dist/macosx" do
       (lib/"python2.7").install "lib/python/site-packages"
+      (lib/"python3.5").install "lib/python3/site-packages" if build.with? :python3
```
Looks great. If you don't mind making one more change, you could automatically obtain the Python version like so:
```RUBY
  def pyver
    Language::Python.major_minor_version "python"
  end
```
and substitute `pyver` where needed, e.g., `(lib/"python#{pyver}").install ...`

Then if you could squash your commits into one, we'll be good to go.

jaguarondi commented:
Changes for python done and commits squashed.

tdsmith commented:
--with-python3 should ideally build for both python2 and python3; is that possible?

dpo commented:
Don't know. @jaguarondi ?

jaguarondi commented:
Yes, that's what I wanted and for me it did install it for both python2 and python3. Unfortunately I moved job and had to return my laptop so I won't be able to test until I get a new Mac, hopefully soon.
From what I remember, the openimageio cmake installs the python bindings if it does detects the corresponding python version.

I added these lines to be able to detect python3 and I think for me it was still able to find python2 at the same time.
```ruby
    # CMake picks up the system's python dylib, even if we have a brewed one.
    # Code taken from the Insighttoolkit formula
    if build.with? "python3"
      ENV["PYTHONPATH"] = lib/"python#{py3ver}/site-packages"
      ENV.append "MY_CMAKE_FLAGS", "-DPYTHON_EXECUTABLE='#{`python3-config --prefix`.chomp}/bin/python3'"
      ENV.append "MY_CMAKE_FLAGS", "-DPYTHON_LIBRARY='#{`python3-config --prefix`.chomp}/lib/libpython#{py3ver}.dylib'"
      ENV.append "MY_CMAKE_FLAGS", "-DPYTHON_INCLUDE_DIR='#{`python3-config --prefix`.chomp}/include/python#{py3ver}m'"
    end
```

tdsmith commented:
looks good with this patch:

```diff
diff --git a/openimageio.rb b/openimageio.rb
index d56f619..7591d0d 100644
--- a/openimageio.rb
+++ b/openimageio.rb
@@ -19,7 +19,6 @@ class Openimageio < Formula
   depends_on "pkg-config" => :build
   depends_on "qt" => :optional # for openimageio viewer
   depends_on "boost"
-  depends_on "boost-python"
   depends_on "openssl"
   depends_on "cfitsio"
   depends_on "field3d"
@@ -38,6 +37,8 @@ class Openimageio < Formula
   depends_on "opencv" => :recommended
   depends_on :python3 => :optional
 
+  depends_on "boost-python" => (build.with?("python3") ? ["with-python3"] : [])
+
   resource "j2kp4files" do
     url "http://pkgs.fedoraproject.org/repo/pkgs/openjpeg/j2kp4files_v1_5.zip/27780ed3254e6eb763ebd718a8ccc340/j2kp4files_v1_5.zip"
     sha256 "21cf3156ed2a2a39765c0d57f36c71d1291e9c30054775a2f0a8fdd2964f1799"
```

tdsmith commented:
@BrewTestBot test this please
